test/new_relic/telemetry_sdk/span_test.rb

```diff
@@ -0,0 +1,115 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-telemetry-sdk-ruby/blob/main/LICENSE for complete details.
+
+require File.expand_path(File.join(File.dirname(__FILE__),'../..','test_helper'))
+
+require 'new_relic/telemetry_sdk/span'
+require 'new_relic/telemetry_sdk/util'
+
+module NewRelic
+  module TelemetrySdk
+    class SpanTest < Minitest::Test
+      def test_required_attributes
+        span = Span.new
+        assert span.id.is_a? String
+        assert span.trace_id.is_a? String
+        assert span.start_time_ms.is_a? Integer
+      end
+
+      def test_recommended_attributes
+        span = Span.new duration_ms: 123456,
+                        parent_id: "c617c2813a222a34",
+                        name: "Name",
+                        service_name: "My Service"
+
+        assert_equal 123456, span.duration_ms
+        assert span.duration_ms.is_a? Integer
+
+        assert_equal "Name", span.name
+        assert span.name.is_a? String
+
+        assert_equal "c617c2813a222a34", span.parent_id
+        assert span.parent_id.is_a? String
+
+        assert_equal "My Service", span.service_name
+        assert span.service_name.is_a? String
+      end
+
+      def test_custom_attributes
+        custom_attributes = {
+          :'user.email' => "me@newr.com",
+          :custom_key   => "custom_value"
+        }
+
+        span = Span.new custom_attributes: custom_attributes
+
+        assert_equal custom_attributes, span.custom_attributes
+      end
```

Comment:
Do we need a test for use-case where a user might add custom_attributes later than during initialization?

Reply:
Yes, you're right. And users should be able to modify at least some other attributes after span creation as well, which is not provided for currently. (I remember you tried to bring this up with me before, and I was saying in the most common case we'd probably be translating already-created spans to the New Relic format, but now that I'm thinking about it again, we definitely want to provide for in-progress spans that can be modified as well.)

Thanks for making this comment. I'll work on modifying this and adding more tests.